### ISSUE TITLE
Creating GUI-based simple questions fails when SQL queries are used as the base data

### ISSUE THREAD

**Issue by KennethLim314** [metabase/metabase#16202]
**Describe the bug**
Trying to run any subsequent summaries on an SQL query results in error
```
Value does not match schema: {:query {:breakout [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}
```

**Logs**
```
[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:49+08:00 INFO metabase.query-processor.middleware.fetch-source-query Fetched source query from Card 178: 
 {:native "SELECT 1 AS val\nUNION ALL\nSELECT 2 AS val\nUNION ALL\nSELECT 3 AS val\nUNION ALL\nSELECT 4 AS val\n"}

[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:49+08:00 INFO metabase.query-processor.middleware.fetch-source-query Fetched source query from Card 178: 
 {:native "SELECT 1 AS val\nUNION ALL\nSELECT 2 AS val\nUNION ALL\nSELECT 3 AS val\nUNION ALL\nSELECT 4 AS val\n"}

[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:50+08:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 828.1 ms (12 DB calls) App DB connections: 1/15 Jetty threads: 2/50 (19 idle, 0 queued) (122 total active threads) Queries in flight: 0 (0 queued)
[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:56+08:00 INFO metabase.api.dataset Source query for this query is Card 178
[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:56+08:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 5,
 :started_at #t "2021-05-24T03:55:56.871301Z[GMT]",
 :error_type :invalid-query,
 :json_query
 {:type "query",
  :query
  {:source-table "card__178", :aggregation [["count"]], :breakout [["field" "val" {:binning {:strategy "default"}}]]},
  :database 5,
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> util.schema$schema_core_validator$fn__17906.invoke(schema.clj:29)"
  "query_processor.middleware.validate$validate_query$fn__50236.invoke(validate.clj:9)"
  "query_processor.middleware.normalize_query$normalize$fn__48117.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__45511.invoke(add_rows_truncated.clj:35)"
  "metabase_enterprise.audit.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__31880.invoke(handle_audit_queries.clj:162)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__50167.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__47362.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__49279.invoke(process_userland_query.clj:135)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__47302.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___33705$thunk__33706.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___33705.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___33714$fn__33717.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___33714.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:239)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:235)"
  "query_processor$fn__50844$process_query_and_save_execution_BANG___50853$fn__50856.invoke(query_processor.clj:251)"
  "query_processor$fn__50844$process_query_and_save_execution_BANG___50853.invoke(query_processor.clj:243)"
  "query_processor$fn__50888$process_query_and_save_with_max_results_constraints_BANG___50897$fn__50900.invoke(query_processor.clj:263)"
  "query_processor$fn__50888$process_query_and_save_with_max_results_constraints_BANG___50897.invoke(query_processor.clj:256)"
  "api.dataset$run_query_async$fn__65655.invoke(dataset.clj:56)"
  "query_processor.streaming$streaming_response_STAR_$fn__65634$fn__65635.invoke(streaming.clj:72)"
  "query_processor.streaming$streaming_response_STAR_$fn__65634.invoke(streaming.clj:71)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__17604.invoke(streaming_response.clj:84)"],
 :context :ad-hoc,
 :error
 "Value does not match schema: {:query {:breakout [(named (not (\":field clauses using a string field name must specify :base-type.\" a-clojure.lang.PersistentVector)) \"Must be a valid instance of one of these clauses: :expression, :field\")]}}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :schema.core/error,
  :value
  {:type :query,
   :query
   {:source-table "card__178", :aggregation [[:count]], :breakout [[:field "val" {:binning {:strategy :default}}]]},
   :database 5,
   :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
   :info
   {:executed-by 1,
    :context :ad-hoc,
    :card-id 178,
    :nested? true,
    :query-hash
    [96, 91, 115, 101, 7, 82, -52, -8, 2, 43, -60, 59, 18, -76, -49, 61, -44, -16, -34, 71, 30, 4, 38, -127, 46, 15,
     -84, -7, 5, 127, -15, -112]},
   :constraints {:max-results 10000, :max-results-bare-rows 2000}},
  :error
  {:query
   {:breakout
    [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}},
 :data {:rows [], :cols []}}

[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:56+08:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 138.1 ms (4 DB calls) App DB connections: 0/15 Jetty threads: 2/50 (18 idle, 0 queued) (121 total active threads) Queries in flight: 0 (0 queued)
[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:57+08:00 INFO metabase.api.dataset Source query for this query is Card 178
[a7d2a8e2-8fbf-4dcb-ae39-ea55387795e7] 2021-05-24T11:55:58+08:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 5,
 :started_at #t "2021-05-24T03:55:57.956100Z[GMT]",
 :error_type :invalid-query,
 :json_query
 {:database 5,
  :query
  {:source-table "card__178", :aggregation [["count"]], :breakout [["field" "val" {:binning {:strategy "default"}}]]},
  :type "query",
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> util.schema$schema_core_validator$fn__17906.invoke(schema.clj:29)"
  "query_processor.middleware.validate$validate_query$fn__50236.invoke(validate.clj:9)"
  "query_processor.middleware.normalize_query$normalize$fn__48117.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__45511.invoke(add_rows_truncated.clj:35)"
  "metabase_enterprise.audit.query_processor.middleware.handle_audit_queries$handle_internal_queries$fn__31880.invoke(handle_audit_queries.clj:162)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__50167.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__47362.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__49279.invoke(process_userland_query.clj:135)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__47302.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___33705$thunk__33706.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___33705.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___33714$fn__33717.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___33714.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:239)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:235)"
  "query_processor$fn__50844$process_query_and_save_execution_BANG___50853$fn__50856.invoke(query_processor.clj:251)"
  "query_processor$fn__50844$process_query_and_save_execution_BANG___50853.invoke(query_processor.clj:243)"
  "query_processor$fn__50888$process_query_and_save_with_max_results_constraints_BANG___50897$fn__50900.invoke(query_processor.clj:263)"
  "query_processor$fn__50888$process_query_and_save_with_max_results_constraints_BANG___50897.invoke(query_processor.clj:256)"
  "api.dataset$run_query_async$fn__65655.invoke(dataset.clj:56)"
  "query_processor.streaming$streaming_response_STAR_$fn__65634$fn__65635.invoke(streaming.clj:72)"
  "query_processor.streaming$streaming_response_STAR_$fn__65634.invoke(streaming.clj:71)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__17604.invoke(streaming_response.clj:84)"],
 :context :ad-hoc,
 :error
 "Value does not match schema: {:query {:breakout [(named (not (\":field clauses using a string field name must specify :base-type.\" a-clojure.lang.PersistentVector)) \"Must be a valid instance of one of these clauses: :expression, :field\")]}}",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data
 {:type :schema.core/error,
  :value
  {:database 5,
   :query
   {:source-table "card__178", :aggregation [[:count]], :breakout [[:field "val" {:binning {:strategy :default}}]]},
   :type :query,
   :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
   :info
   {:executed-by 1,
    :context :ad-hoc,
    :card-id 178,
    :nested? true,
    :query-hash
    [96, 91, 115, 101, 7, 82, -52, -8, 2, 43, -60, 59, 18, -76, -49, 61, -44, -16, -34, 71, 30, 4, 38, -127, 46, 15,
     -84, -7, 5, 127, -15, -112]},
   :constraints {:max-results 10000, :max-results-bare-rows 2000}},
  :error
  {:query
   {:breakout
    [(named (not (":field clauses using a string field name must specify :base-type." a-clojure.lang.PersistentVector)) "Must be a valid instance of one of these clauses: :expression, :field")]}}},
 :data {:rows [], :cols []}}
```

**To Reproduce**
create initial table as follows and save as question:
```
SELECT 1 AS val
UNION ALL
SELECT 2 AS val
UNION ALL
SELECT 3 AS val
UNION ALL
SELECT 4 AS val
```
Then create a question using the above table as base dataset and run a count over the SQL question

**Expected behavior**
Counts the number of rows, returning 4

**Screenshots**

**Information about your Metabase Installation:**

{
  "browser-info": {
    "language": "en-GB",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.11+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.11",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.11+9",
    "os.name": "Linux",
    "os.version": "4.14.219-164.354.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "bigquery"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-04-27",
      "tag": "v1.39.1",
      "branch": "release-x.39.x",
      "hash": "6beba48"
    },
    "settings": {
      "report-timezone": null
    }
  }
}

**Severity**
Annoying

**Additional context**
Previously worked, might be a regression

**Comment by KennethLim314** [comment]
This same error, or at least something similar, seems to be happening if you perform any aggregation using another dataset as a data source

**Comment by flamber** [comment]
Closing in favor of #15397, which was fixed by #16140 and will be part of 0.39.3
